src/shogun/distributions/HMM.cpp

```diff
@@ -7,19 +7,21 @@
  */
 #include <shogun/distributions/HMM.h>
 #include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/RandomNamespace.h>
 #include <shogun/io/SGIO.h>
 #include <shogun/lib/config.h>
 #include <shogun/lib/Signal.h>
 #include <shogun/base/Parallel.h>
 #include <shogun/features/StringFeatures.h>
 #include <shogun/features/Alphabet.h>
+#include <shogun/mathematics/UniformRealDistribution.h>
 
 #include <stdlib.h>
 #include <stdio.h>
 #include <time.h>
 #include <ctype.h>
 
-#define VAL_MACRO log((default_value == 0) ? (CMath::random(MIN_RAND, MAX_RAND)) : default_value)
+#define VAL_MACRO log((default_value == 0) ? (uniform_real_dist(m_prng)) : default_value)
```

Comment:
this doesnt seem to be the same?

Reply:
@karlnapf they are, but the limits (MIN_RAND, MAX_RAND) are set at construction site of uniform_real_dist